client/service/src/client/call_executor.rs

```diff
@@ -127,11 +126,6 @@ where
 		extensions: Option<Extensions>,
 	) -> sp_blockchain::Result<Vec<u8>> {
 		let mut changes = OverlayedChanges::default();
-		let mut offchain_changes = if self.client_config.offchain_indexing_api {
```

Comment:
should we really disable the offchain indexing don't feed change optimitization?
If it was huge gain.

Reply:
I think we could open an issue to keep track of this. Currently it only means that we write to the indexing overelay, but discard it when we apply it to the backend when indexing is disabled.

Reply:
The point is, we call to the host anyway and this is probably the expensive part when doing this call. Doing some host allocation/storing in hashmap is probably negligible. But yeah, this is just some "feeling"